lib/statsd_middleware.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+class StatsdMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    status, headers, response = @app.call(env)
+    params = env['action_dispatch.request.path_parameters']
+    key = "api.external.request#status=#{status},controller=#{params[:controller]},action=#{params[:action]}"
```

Comment:
Oh nice, I like the use of controller + action as the key vs uri. I'm curious to test out some negative tests with this though, such as ActionController::RoutingError